src/main/java/io/excaliburfrc/robot/subsystems/Climber.java

```diff
@@ -37,6 +38,25 @@
   private final SparkMaxPIDController leftController = leftMotor.getPIDController();
   private final SparkMaxPIDController rightController = rightMotor.getPIDController();
 
+  private final ElevatorFeedforward upFF = new ElevatorFeedforward(kS, MG, kV, kA);
+  private final TrapezoidProfile.Constraints upConstraints =
+      new TrapezoidProfile.Constraints(
+          upFF.maxAchievableVelocity(12, 1), upFF.maxAchievableAcceleration(12, 1));
+
+  private final ElevatorFeedforward diagonalFF =
+      new ElevatorFeedforward(kS, MG * Math.cos(ANGLE), kV, kA);
+  private final TrapezoidProfile.Constraints diagonalConstraints =
+      new TrapezoidProfile.Constraints(
+          diagonalFF.maxAchievableVelocity(12, 1), diagonalFF.maxAchievableAcceleration(12, 1));
+
+  private final TrapezoidProfile fullUpProfile =
+      new TrapezoidProfile(upConstraints, new TrapezoidProfile.State(FULL_UP_HEIGHT, 0));
+  private final TrapezoidProfile someUpProfile =
+      new TrapezoidProfile(upConstraints, new TrapezoidProfile.State(SOME_UP_HEIGHT, 0));
+  private final TrapezoidProfile diagonalProfile =
+      new TrapezoidProfile(
+          diagonalConstraints, new TrapezoidProfile.State(FULL_UP_HEIGHT - SOME_UP_HEIGHT, 0));
```

Comment:
I see that you pass one state into each of these. How does that work?

Reply:
The init is 0, 0

Reply:
that's the default?

even if it is, passing a `(0,0)` state explicitly is expressive. and `diagonalProfile` shouldn't be starting at 0 anyway.

additionally, those two profiles can be merged, along with triggering the angle offset once the position nears a certain point.

Reply:
Which two profiles?
And what do you mean by merge?
We can set one's goal to a certain velocity and set the other's init to the same velocity if that's what you mean